### PR TITLE
Fork calculating the health indicators to Management pool

### LINKED ISSUE #91597: [CI] HealthRestCancellationIT testHealthRestCancellation failing
This looks to be a newly added test that has failed more than 50% of the time in CI. This also reproduces locally for me.

**Build scan:**
https://gradle-enterprise.elastic.co/s/cog5ggkrt7mr2/tests/:qa:smoke-test-http:javaRestTest/org.elasticsearch.http.HealthRestCancellationIT/testHealthRestCancellation

**Reproduction line:**
```
./gradlew ':qa:smoke-test-http:javaRestTest' --tests "org.elasticsearch.http.HealthRestCancellationIT.testHealthRestCancellation" -Dtests.seed=E6DB242184DD5DF3 -Dtests.locale=es-CR -Dtests.timezone=Africa/Porto-Novo -Druntime.java=17
```

**Applicable branches:**
main

**Reproduces locally?:**
Yes

**Failure history:**
https://gradle-enterprise.elastic.co/scans/tests?tests.container=org.elasticsearch.http.HealthRestCancellationIT&tests.test=testHealthRestCancellation

**Failure excerpt:**
```
junit.framework.AssertionFailedError: task 55/cluster:monitor/health_api not cancelled

  at __randomizedtesting.SeedInfo.seed([E6DB242184DD5DF3:916B242BC35F94B5]:0)
  at junit.framework.Assert.fail(Assert.java:57)
  at junit.framework.Assert.assertTrue(Assert.java:22)
  at junit.framework.TestCase.assertTrue(TestCase.java:192)
  at org.elasticsearch.test.TaskAssertions.lambda$assertAllCancellableTasksAreCancelled$2(TaskAssertions.java:66)
  at org.elasticsearch.test.ESTestCase.assertBusy(ESTestCase.java:1112)
  at org.elasticsearch.test.TaskAssertions.assertAllCancellableTasksAreCancelled(TaskAssertions.java:57)
  at org.elasticsearch.http.HealthRestCancellationIT.runTest(HealthRestCancellationIT.java:82)
  at org.elasticsearch.http.HealthRestCancellationIT.testHealthRestCancellation(HealthRestCancellationIT.java:50)
  at jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(NativeMethodAccessorImpl.java:-2)
  at jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:77)
  at jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
  at java.lang.reflect.Method.invoke(Method.java:568)
  at com.carrotsearch.randomizedtesting.RandomizedRunner.invoke(RandomizedRunner.java:1758)
  at com.carrotsearch.randomizedtesting.RandomizedRunner$8.evaluate(RandomizedRunner.java:946)
  at com.carrotsearch.randomizedtesting.RandomizedRunner$9.evaluate(RandomizedRunner.java:982)
  at com.carrotsearch.randomizedtesting.RandomizedRunner$10.evaluate(RandomizedRunner.java:996)
  at com.carrotsearch.randomizedtesting.rules.StatementAdapter.evaluate(StatementAdapter.java:36)
  at org.apache.lucene.tests.util.TestRuleSetupTeardownChained$1.evaluate(TestRuleSetupTeardownChained.java:44)
  at org.apache.lucene.tests.util.AbstractBeforeAfterRule$1.evaluate(AbstractBeforeAfterRule.java:43)
  at org.apache.lucene.tests.util.TestRuleThreadAndTestName$1.evaluate(TestRuleThreadAndTestName.java:45)
  at org.apache.lucene.tests.util.TestRuleIgnoreAfterMaxFailures$1.evaluate(TestRuleIgnoreAfterMaxFailures.java:60)
  at org.apache.lucene.tests.util.TestRuleMarkFailure$1.evaluate(TestRuleMarkFailure.java:44)
  at com.carrotsearch.randomizedtesting.rules.StatementAdapter.evaluate(StatementAdapter.java:36)
  at com.carrotsearch.randomizedtesting.ThreadLeakControl$StatementRunner.run(ThreadLeakControl.java:390)
  at com.carrotsearch.randomizedtesting.ThreadLeakControl.forkTimeoutingTask(ThreadLeakControl.java:843)
  at com.carrotsearch.randomizedtesting.ThreadLeakControl$3.evaluate(ThreadLeakControl.java:490)
  at com.carrotsearch.randomizedtesting.RandomizedRunner.runSingleTest(RandomizedRunner.java:955)
  at com.carrotsearch.randomizedtesting.RandomizedRunner$5.evaluate(RandomizedRunner.java:840)
  at com.carrotsearch.randomizedtesting.RandomizedRunner$6.evaluate(RandomizedRunner.java:891)
  at com.carrotsearch.randomizedtesting.RandomizedRunner$7.evaluate(RandomizedRunner.java:902)
  at org.apache.lucene.tests.util.AbstractBeforeAfterRule$1.evaluate(AbstractBeforeAfterRule.java:43)
  at com.carrotsearch.randomizedtesting.rules.StatementAdapter.evaluate(StatementAdapter.java:36)
  at org.apache.lucene.tests.util.TestRuleStoreClassName$1.evaluate(TestRuleStoreClassName.java:38)
  at com.carrotsearch.randomizedtesting.rules.NoShadowingOrOverridesOnMethodsRule$1.evaluate(NoShadowingOrOverridesOnMethodsRule.java:40)
  at com.carrotsearch.randomizedtesting.rules.NoShadowingOrOverridesOnMethodsRule$1.evaluate(NoShadowingOrOverridesOnMethodsRule.java:40)
  at com.carrotsearch.randomizedtesting.rules.StatementAdapter.evaluate(StatementAdapter.java:36)
  at com.carrotsearch.randomizedtesting.rules.StatementAdapter.evaluate(StatementAdapter.java:36)
  at org.apache.lucene.tests.util.TestRuleAssertionsRequired$1.evaluate(TestRuleAssertionsRequired.java:53)
  at org.apache.lucene.tests.util.AbstractBeforeAfterRule$1.evaluate(AbstractBeforeAfterRule.java:43)
  at org.apache.lucene.tests.util.TestRuleMarkFailure$1.evaluate(TestRuleMarkFailure.java:44)
  at org.apache.lucene.tests.util.TestRuleIgnoreAfterMaxFailures$1.evaluate(TestRuleIgnoreAfterMaxFailures.java:60)
  at org.apache.lucene.tests.util.TestRuleIgnoreTestSuites$1.evaluate(TestRuleIgnoreTestSuites.java:47)
  at com.carrotsearch.randomizedtesting.rules.StatementAdapter.evaluate(StatementAdapter.java:36)
  at com.carrotsearch.randomizedtesting.ThreadLeakControl$StatementRunner.run(ThreadLeakControl.java:390)
  at com.carrotsearch.randomizedtesting.ThreadLeakControl.lambda$forkTimeoutingTask$0(ThreadLeakControl.java:850)
  at java.lang.Thread.run(Thread.java:833)

```

### PR DESCRIPTION
**Pull request by andreidan**
When a request for the health API comes in we first reach out to the health node (transport request) to fetch the cluster health state. When we get a response we calculate all the health indicators (except `master_is_stable` which executes before we do anything else).

Calculating the health indicators can get expensive when a large number of shards are present in the cluster and we shouldn't keep the transport worker thread busy with this.

This forks the health indicators' health computation to the Management pool.

Fixes #91597

### CODE REVIEW

File: server/src/test/java/org/elasticsearch/health/HealthServiceTests.java [hunk 1]
```diff
@@ -94,30 +112,6 @@ public void onFailure(Exception e) {
         };
     }
 
-    public void testDuplicateIndicatorNames() throws Exception {
```
Comment (andreidan) [inline server/src/test/java/org/elasticsearch/health/HealthServiceTests.java:97]: Removed this test as I don’t think we should test assertions (if they’re not assertions they should be failures) and I don’t think we should surface assertions in the async listener.

Comment (masseyke) [reply server/src/test/java/org/elasticsearch/health/HealthServiceTests.java:97]: Is this related to this PR? I don't think I see the connection.

File: server/src/main/java/org/elasticsearch/health/HealthService.java [hunk 2]
```diff
@@ -105,24 +108,53 @@ public void getHealth(
...
                         indicatorName,
-                        Stream.concat(filteredPreflightResults, filteredIndicators.map(service -> service.calculate(explain, healthInfo)))
-                            .toList(),
+                        healthInfo,
+                        explain,
                         listener
                     );
+
+                    try {
+                        threadPool.executor(ThreadPool.Names.MANAGEMENT).submit(calculateFilteredIndicatorsRunnable);
```
Comment (masseyke) [inline server/src/main/java/org/elasticsearch/health/HealthService.java:121]: Is management the right thread pool here? Our docs say that's `For cluster management.` I'm not sure this counts as that, does it (or maybe it does -- I could interpret that either way)? Would `generic` be better?

**Comment by andreidan** [comment]
@elasticmachine update branch

File: server/src/main/java/org/elasticsearch/health/HealthService.java (hunk 2 above)
Comment (andreidan) [reply server/src/main/java/org/elasticsearch/health/HealthService.java:121]: I think health it does have to do with the management of the cluster/data/deployment. Generic is quite vague IMO.
Do you have a strong opinion here?

`_cluster/health` goes to `management` too if that matters at all

File: server/src/test/java/org/elasticsearch/health/HealthServiceTests.java (hunk 1 above)
Comment (andreidan) [reply server/src/test/java/org/elasticsearch/health/HealthServiceTests.java:97]: This assertion is now in a background thread https://github.com/elastic/elasticsearch/pull/91628/files#diff-444fdb9629cdf9a3ece311d4e8ecd627777532a0f878afc2c7aac5cc3e8ebd6aR190

Irrespective of that, I don't think we should be testing the assertions (if it's not an invariant I think it should be a failure and that failure should be tested via the listener.onFailure thing). I don't think that's the case here though.

File: server/src/main/java/org/elasticsearch/health/HealthService.java (hunk 2 above)
Comment (masseyke) [reply server/src/main/java/org/elasticsearch/health/HealthService.java:121]: If `_cluster/health` does it, I guess that's precedent enough.

File: server/src/test/java/org/elasticsearch/health/HealthServiceTests.java (hunk 1 above)
Comment (masseyke) [reply server/src/test/java/org/elasticsearch/health/HealthServiceTests.java:97]: OK I see the reason for removing it -- it's an Error in the background thread, and we only catch Exceptions. So we're going to lose test coverage for `findDuplicatesByName()`. But I guess that's not a huge loss.

**Review by masseyke — APPROVED** [review]
LGTM. I've run HealthRestCancellationIT.testHealthRestCancellation() several hundred times now with no failures.

**Pull request merged**